vscode-dotnet-runtime-library/src/Acquisition/DotnetConditionValidator.ts

```diff
@@ -0,0 +1,150 @@
+{
+    public constructor(private readonly workerContext : IAcquisitionWorkerContext, private readonly utilityContext : IUtilityContext, private executor? : ICommandExecutor)
+    {
+        this.executor ??= new CommandExecutor(this.workerContext, this.utilityContext);
+    }
+
+    public async dotnetMeetsRequirement(dotnetExecutablePath: string, requirement : IDotnetFindPathContext) : Promise<boolean>
+    {
+        const availableRuntimes = await this.getRuntimes(dotnetExecutablePath);
+        const requestedMajorMinor = versionUtils.getMajorMinor(requirement.acquireContext.version, this.workerContext.eventStream, this.workerContext);
+        const hostArch = await this.getHostArchitecture(dotnetExecutablePath);
+
+        if(availableRuntimes.some((runtime) =>
+            {
+                const foundVersion = versionUtils.getMajorMinor(runtime.version, this.workerContext.eventStream, this.workerContext);
+                return runtime.mode === requirement.acquireContext.mode && this.stringArchitectureMeetsRequirement(hostArch, requirement.acquireContext.architecture) &&
+                    this.stringVersionMeetsRequirement(foundVersion, requestedMajorMinor, requirement.versionSpecRequirement);
+            }))
+        {
+            return true;
+        }
+        else
+        {
+            const availableSDKs = await this.getSDKs(dotnetExecutablePath);
+            if(availableSDKs.some((sdk) =>
+                {
+                    // The SDK includes the Runtime, ASP.NET Core Runtime, and Windows Desktop Runtime. So, we don't need to check the mode.
+                    const foundVersion = versionUtils.getMajorMinor(sdk.version, this.workerContext.eventStream, this.workerContext);
+                    return this.stringArchitectureMeetsRequirement(hostArch, requirement.acquireContext.architecture), this.stringVersionMeetsRequirement(foundVersion, requestedMajorMinor, requirement.versionSpecRequirement);
+                }))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     *
+     * @param hostPath The path to the dotnet executable
+     * @returns The architecture of the dotnet host from the PATH, in dotnet info string format
+     * The .NET Host will only list versions of the runtime and sdk that match its architecture.
+     * Thus, any runtime or sdk that it prints out will be the same architecture as the host.
+     *
+     * @remarks Will return '' if the architecture cannot be determined for some peculiar reason (e.g. dotnet --info is broken or changed).
+     */
+    // eslint-disable-next-line @typescript-eslint/require-await
+    private async getHostArchitecture(hostPath : string) : Promise<string>
+    {
+        return '';
+        /* The host architecture can be inaccurate. Imagine a local runtime install with no host. There is no way to tell the architecture of that runtime,

```

Comment:
the host is always included with the runtime, doesn't that make the installation incomplete and therefore invalid?

Reply:
That's true, this comment (the one in the code) is wrong. However, the host can be there but not return an architecture if it's just a local runtime and a host. I should update this comment.